serverless/functions/serverless-functions-getting-started.adoc

```diff
@@ -6,11 +6,11 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
+This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation. When building and deploying functions, the resulting container image is stored in an image registry. Usually this is a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.
```

Comment:
I guess the only thing I would say is that this is a big "if". It's entirely likely that _most_ OpenShift administrators haven't exposed the internal registry in this way - especially not for junior or mid level developers, who could easily be the kind of developers writing functions. If this is the case, and they try to deploy a function without the proper privileges, they will get an error. I think it might be important to add some cautionary text, e.g.

> **Warning**
> If you are logged in to an OpenShift cluster and do not have authorization for the Container Registry, you will need to configure your default image registry using the `FUNC_REGISTRY` environment variable.

Or something to that effect but better. ;)

Reply:
The registry does not need to be exposed to outer internet and it is used by default now (for OpenShift clusters).

>Usually this is a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.

Should be more like:
>By default the internal OpenShift registry is used,  this behariour can be overridden by the `-r` flag.

Reply:
> The registry does not need to be exposed to outer internet and it is used by default now (for OpenShift clusters).

Hm OK. I feel like I have run into errors before if I'm logged into an openshift registry. Or maybe it's just picking up an old kubecontext for a cluster I don't have around anymore. 🤷‍♂️

Reply:
I think @rhuss onece saw such a failure. 

I tested this with `developer` account on CRC and in non-admin in Sandbox and it works.
Also @jrangelramos  tested this with non-admin in some devcluster and it worked for him.

Reply:
@lance if you can reproduce the auth error please let me know how.

Reply:
> By default the internal OpenShift registry is used, this behariour can be overridden by the -r flag (or FUNC_REGISTRY envvar).

I am very confused at this point; are we removing the `-r` flag or not?

Maybe we can remove this text entirely since we have [this section](https://docs.openshift.com/container-platform/4.10/serverless/functions/serverless-functions-setup.html#prerequisites_serverless-functions-setup) already? Surely if they follow the docs for setting up as we request them to at the beginning of the getting started, none of this should be an issue anyway?

Reply:
>It's entirely likely that most OpenShift administrators haven't exposed the internal registry in this way - especially not for junior or mid level developers, who could easily be the kind of developers writing functions. If this is the case, and they try to deploy a function without the proper privileges, they will get an error. I think it might be important to add some cautionary text, e.g.

In that case I would recommend not even mentioning it as an option to be totally honest. That is a very big "IF" and I'm not sure why it would be useful to call it out as an option if realistically it's probably not an option for most devs?

I think in general we need more thought going into the E2E user stories for some of this set up stuff for functions. It feels like we keep running into issues when we come to the point of trying to figure out how to document something because we're not sure how it's supposed to work / what the recommended use cases are, or whether they can even be achieved by users in reality.

Reply:
@abrennan89 sorry to have caused a ruckus. Please disregard my comment. I think this PR looks good.

Reply:
@matejvasek comment is a nice improvement though.
```suggestion
This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation. When building and deploying functions, the resulting container image is stored in an image registry. This can be a public registry, such as Quay or the OpenShift Container Registry. By default the OpenShift Container Registry is used, this behavior can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
```

Reply:
What Lance suggested above looks good.

The `-r` flag is also mentioned here: https://docs.openshift.com/container-platform/4.10/serverless/functions/serverless-functions-getting-started.html#serverless-build-func-kn_serverless-functions-getting-started
It states that is has to be set which is not true anymore.
We might remove notion of the `-r` flag from there since it is not mandatory.

Reply:
```diff
diff --git a/modules/serverless-build-func-kn.adoc b/modules/serverless-build-func-kn.adoc
index 41fb395a2c..ae32d63596 100644
--- a/modules/serverless-build-func-kn.adoc
+++ b/modules/serverless-build-func-kn.adoc
@@ -8,13 +8,15 @@
 
 Before you can run a function, you must build the function project by using the `kn func build` command. The build command reads the `func.yaml` file from the function project directory to determine the image name and registry.
 
+By default the OpenShift Container Registry is used in `func.yaml` when creating new project. The registry used to store Function images can be overridden using the `-r` flag or the `FUNC_REGISTRY` environment variable.
+
 .Example `func.yaml`
 [source,yaml]
 ----
 name: example-function
 namespace: default
 runtime: node
-image: <image_from_registry>
+image: image-registry.openshift-image-registry.svc:5000/<oc-project>/<func-name>
 imageDigest: ""
 trigger: http
 builder: default
@@ -23,8 +25,6 @@ builderMap:
 envs: {}
 ----
 
-If the image name and registry are not set in the `func.yaml` file, you must either specify the registry flag, `-r` when using the `kn func build` command, or you are prompted to provide a registry value in the terminal when building a function. An image name is then derived from the registry value that you have provided.
-
 .Example command using the `-r` registry flag
 [source,terminal]
 ----
diff --git a/serverless/functions/serverless-functions-getting-started.adoc b/serverless/functions/serverless-functions-getting-started.adoc
index 135b281bf7..9bbad7e3cf 100644
--- a/serverless/functions/serverless-functions-getting-started.adoc
+++ b/serverless/functions/serverless-functions-getting-started.adoc
@@ -6,7 +6,7 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
-This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation. When building and deploying functions, the resulting container image is stored in an image registry. Usually this is a public registry, such as Quay. However, you can use the integrated OpenShift Container Registry instead if it has been exposed by a cluster administrator.
+This guide explains how you can get started with creating, building, and deploying a function on an {ServerlessProductName} installation.
 
 :FeatureName: {FunctionsProductName}
 include::snippets/technology-preview.adoc[leveloffset=+2]
 ```